lib/workers/repository/process/lookup/index.ts

```diff
@@ -537,7 +537,13 @@ export async function lookupUpdates(
           const getDigestConfig: GetDigestInputConfig = {
             ...config,
             registryUrl: update.registryUrl ?? res.registryUrl,
-            lookupName: res.lookupName,
+            // TODO #27728
+            lookupName:
+              update.updateType === 'replacement' ? undefined : res.lookupName,
```

Comment:
Should we also check whether there's a replacement name? e.g. if no replacement name then is it ok to keep using res.lookupName?

Reply:
A replacement update is created when a replacementName, replacementNameTemplate or replacementVersion is defined. 

The replacement update is created here:
https://github.com/renovatebot/renovate/blob/25255596d63a03a312885aba1b25fdfd7b76c7a4/lib/workers/repository/process/lookup/index.ts#L472

And this function validates if it's needed:
https://github.com/renovatebot/renovate/blob/25255596d63a03a312885aba1b25fdfd7b76c7a4/lib/workers/repository/process/lookup/utils.ts#L28

The only case when it may make sense to keep the lookupName if it's only the version is updated, but despite that it seems that the value is calculated again in the getDigest function because if the lookupName doesn't exist uses the packageName (the lookupName is based on the packageName)

https://github.com/renovatebot/renovate/blob/25255596d63a03a312885aba1b25fdfd7b76c7a4/lib/modules/datasource/docker/index.ts#L811